### PR TITLE
Fetch information from confuguration json file

### LINKED ISSUE #111: Create ns-vue-radio tool: Fetch information from file
Depends on #109 

Implement necessary code to ~fetch information from an API~ read data from a file, ~write it's content to a sample file with any name~. The script should read the information and declare an environment variable with a stringified json containing that data. Then, this data must be read from the webpack configuration and declared alongside with API_URL, when it should be JSON parsed. Finally, this variables are used in config.js to populate the attributes.

The script should continue until it succesfully finishes the build.

For this issue we'll need to create a dummy file to read, starting with this attributes:

**Json format**

### name

+   Details: The name of the radio
+   Type: `String`  

### social

+   Details: An object with social media information
+   Type: `Object`
+   Structure:
    +   facebookUrl: `String`
    +   instagramUrl: `String`
    +   twitterUrl: `String`

### colors

+   Details: An object with colors data
+   Type: `Object`
+   Structure:
    +   backgroundColor: `String: hex color`
    +   primaryText: `String: hex color`
    +   secondaryText: `String: hex color`

### bottomNavigationBar

+   Details: An object with data for the bottom navigation bar
+   Type: `Object`
+   Structure:
    +   activeColor: `String: hex color`
    +   inactiveColor: `String: hex color`
    +   backgroundColor: `String: hex color`

### streamingUrl

+   Details: the url to the live streaming
+   Type: `String`

### PR DESCRIPTION
**Pull request by estefi-prieto**
This PR provides:
tool to read information from configuration file
use the configuration data from file with default values too

Closes #111

### CODE REVIEW

File: tools/create-app.bash [hunk 1]
```diff
@@ -9,12 +9,21 @@ KEYSTORE_PATH=$4
...
+### Script path and directory ###
 
-# Place logo in the assets directory
 FULL_PATH=$(realpath $0)
 DIRECTORY_PATH=$(dirname $FULL_PATH)
+
+### Initial configuration ###
+
+# Parse json configuration file to string
+STRING_CONFIGURATION_FILE=$(jq '. | tostring' $CONFIGURATION_FILE)
```
Comment (sgobotta) [inline tools/create-app.bash]: if it's a string, it's not a file anymore : ) we can just use `STRINGIFIED_CONFIGURATION` as variable.

I get the tostring part of the script but, what does `jq` means?

File: tools/create-app.bash [hunk 2]
```diff
@@ -25,8 +34,5 @@ tns resources generate icons $ASSETS_DIRECTORY/icon.png
 
 ### Build ###
 
-# Sign app for release
-tns build android --bundle --release --compileSdk 28 --key-store-path $KEYSTORE_PATH --key-store-password $KEYSTORE_PASS --key-store-alias $KEYSTORE_ALIAS --key-store-alias-password $KEYSTORE_ALIAS_PASS
-
 # Generate a .aab file
-tns build android --bundle --release --compileSdk 28 --key-store-path $KEYSTORE_PATH --key-store-password $KEYSTORE_PASS --key-store-alias $KEYSTORE_ALIAS --key-store-alias-password $KEYSTORE_ALIAS_PASS --aab --copy-to $OUTPUT_DIRECTORY
+tns build android --bundle --release --env.customization=$STRING_CONFIGURATION_FILE --compileSdk 28 --key-store-path $KEYSTORE_PATH --key-store-password $KEYSTORE_PASS --key-store-alias $KEYSTORE_ALIAS --key-store-alias-password $KEYSTORE_ALIAS_PASS --aab --copy-to $OUTPUT_DIRECTORY
```
Comment (sgobotta) [inline tools/create-app.bash]: What do you think about using uppercase for env variables?

File: webpack.config.js [hunk 3]
```diff
@@ -245,7 +246,8 @@ module.exports = env => {
                 "global.TNS_WEBPACK": "true",
                 "TNS_ENV": JSON.stringify(mode),
                 "process": "global.process",
-                "API_URL": JSON.stringify(api_url)
+                "API_URL": JSON.stringify(api_url),
+                "customization": customization.split('\\').join('')
```
Comment (sgobotta) [inline webpack.config.js]: Can we use uppercase for env variables, just like the other custom variables? (TNS_ENV, API_URL)

File: app/config.js [hunk 4]
```diff
@@ -2,29 +2,29 @@ const config = {
   'jsondata': true,
```
Comment (sgobotta) [inline app/config.js:23]: We can avoid code repetition if we use destructuring ES6 syntax, like we did in some vue components.

File: app/config.js [hunk 5]
```diff
@@ -2,29 +2,29 @@ const config = {
...
+      color: customization.colors.secondaryText || '#ffffff'
     }
   },
   'phoneNumber': '',
   'shareApp': {
     'playStoreUrl': '',
     'message': ''
   },
-  'sections': [
+  'sections': customization.sections || [
```
Comment (sgobotta) [inline app/config.js]: Why we don't need Social and and Podcasts by default anymore?

File: .eslintrc [hunk 6]
```diff
@@ -20,7 +20,8 @@
   ],
   "globals": {
     "API_URL": true,
-    "TNS_ENV": true
+    "TNS_ENV": true,
+    "customization": true
```
Comment (sgobotta) [inline .eslintrc]: If we change it to uppercase we'd have to edit this too I  think.

**Review by sgobotta — CHANGES_REQUESTED** [review]

File: tools/create-app.bash (hunk 2 above)
Comment (estefi-prieto) [reply tools/create-app.bash]: It's the same, but if the rest of the variables of the webpack use lowercase, we should use the same convention.

File: tools/create-app.bash (hunk 1 above)
Comment (estefi-prieto) [reply tools/create-app.bash]: > I get the tostring part of the script but, what does jq means?

it's a command-line JSON processor.

[jq manual](https://stedolan.github.io/jq/manual/)

File: webpack.config.js (hunk 3 above)
Comment (estefi-prieto) [reply webpack.config.js]: It was changed

File: app/config.js (hunk 4 above)
Comment (estefi-prieto) [reply app/config.js:23]: I think the same. I changed it

File: .eslintrc (hunk 6 above)
Comment (estefi-prieto) [reply .eslintrc]: It was changed too

File: tools/create-app.bash (hunk 1 above)
Comment (sgobotta) [reply tools/create-app.bash]: Alrighty, good to know : )

File: tools/create-app.bash (hunk 2 above)
Comment (sgobotta) [reply tools/create-app.bash]: Yes, you're right, I agree with you, let's keep the convention webpack already uses.

File: app/config.js [hunk 7]
```diff
@@ -1,30 +1,51 @@
+const {
+  bottomNavigationBar: {
+    activeColor: navActiveColor,
```
Comment (sgobotta) [inline app/config.js:3]: loved this aliases >:D

**Review by sgobotta — APPROVED** [review]

**Comment by sgobotta** [comment]
Nice job, good to merge!!! :tada:

**Pull request merged**